### PR TITLE
Update src/readme.md to include prerequisites 

### PR DESCRIPTION
**Pull request by AlgoEddy**
Just went through the compilation steps for the first time and ran into an error from not having a C++ compiler installed. I added a line to the readme so others don't have a similar issue.

### CODE REVIEW

File: src/README.md [hunk 1]
```diff
@@ -25,6 +25,11 @@ translate the command line arguments.
 See the output of `./build.d --help` to get a list of all supported targets
 or see the list of major targets below.
 
+### Prerequisites
+
+To bootstrap `dmd` you need to have `make` and a `C++` compiler 
+installed, such as gcc.
```
Comment (MoonlightSentinel) [inline src/README.md:31]: These requirements are not true for dmd by itself: 

> make

Executing `rdmd src/build.d` should work on it's own. The legacy makefiles simply forward to that script.

> C++ compiler

Only for POSIX platforms where dmd uses `$CC` / `cc` as a linker. It'll use the bundled `lld-link` on Windows if VS is not installed.

**Review by MoonlightSentinel — COMMENTED** [review]
Thanks for your PR!

File: src/README.md (hunk 1 above)
Comment (AlgoEddy) [reply src/README.md:31]: Okay, we should update some of the build instructions elsewhere that say you should run `make -f posix.mak` then! I'll take a look in the morning and add them to this PR, and amend this one to be correct :)

Comment (AlgoEddy) [reply src/README.md]: @MoonlightSentinel Okay I updated the readme to reflect your comments. The place I found the outdated build instructions is here - https://dlang.org/dmd-osx.html#compiling-dmd, so I can't include it in the PR!

Comment (ibuclaw) [reply src/README.md]: A C++ compiler is only needed if you want to run the unit-tests, so maybe build.d can be fixed to defer checking `$CXX` until it knows that it'll need it?

Comment (ibuclaw) [reply src/README.md]: You do require gcc for linking though.

Comment (MoonlightSentinel) [reply src/README.md]: The requirement of `build.d` on  `CXX` is fixed by #13523

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**